src/components/workspace/side-content/Autograder.tsx

```diff
@@ -1,28 +1,85 @@
+import { Collapse } from '@blueprintjs/core';
+import { IconNames } from '@blueprintjs/icons';
 import * as React from 'react';
-import { ITestcase } from '../../assessment/assessmentShape';
+import { AutogradingResult, ITestcase } from '../../assessment/assessmentShape';
+import { controlButton } from '../../commons';
 import AutograderCard from './AutograderCard';
+import ResultCard from './ResultCard';
 
 type AutograderProps = {
+  autogradingResults: AutogradingResult[];
   testcases: ITestcase[] | null;
   handleTestcaseEval: (testcaseId: number) => void;
 };
 
-class Autograder extends React.Component<AutograderProps, {}> {
+type State = {
+  showTestcases: boolean;
+  showResults: boolean;
+};
+
+class Autograder extends React.Component<AutograderProps, State> {
+  public constructor(props: AutograderProps) {
+    super(props);
+
+    const hasResults: boolean = this.props.autogradingResults.length > 0 ? true : false;
```

Comment:
`const hasResults: boolean = this.props.autogradingResults.length > 0;`

Reply:
Refactored